### ISSUE TITLE
Preinstall of AppStore add-in in WOPI session failing

### ISSUE THREAD

**Issue by nifty-nev** [OfficeDev/office-js#5016]
# Provide required information needed to triage your issue
This issue is preventing us from rolling our new word add-in out to customers.  The impact on our business is high.
## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: WOPI session running in desktop browser (add-in preinstall tried and failed with Edge, Chrome and Firefox).  The add-in works in the native desktop application and when side loaded to a WOPI session
* Host [Excel, Word, PowerPoint, etc.]: Word
* Office version number: 16
* Operating System: Windows 11 for client
* Browser (if using Office on the web): Chrome 130.0.6723.70, Edge 130.0.2849.56, Firefox 131.0.3
 
## Expected behavior

After publishing our add-in to the MS AppStore and following the procedure to confirm that the add-in's top frame is trusted (that is, complying with SFTC) and adding a form input field to define the value of host_install_addins it is expected that the add-in will preinstall upon opening of a word document in one of our WOPI sessions.
## Current behavior

The add-in does not appear to preinstall or is not visible to the user if it does

There should be a visible sign that the add-in has preinstalled and is ready for use
## Steps to reproduce

1. A test account and instructions can be provided via a side channel upon request
2. ______
3. ______

## Link to live example(s)

1. A link to a live example on our system can be provided via a side channel upon request
2. ______
3. ______

# Provide additional details

1. ______
2. ______
3. ______

## Context

Our SaaS customers should be able to use our add-in to auto-populate content controls located in corporate documents using values for which our system is their source-of-truth

## Useful logs

- [ ] Console errors
[steps (2).zip](https://github.com/user-attachments/files/17552779/steps.2.zip)

- [ ] Screenshots see attached zip
- [ ] Test file (if only happens on a particular file) See attached zip

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by RuizhiSunMS** [comment]
hi @nifty-nev, thx for your submit. Would you please help confirm something?
1. your addin was installed successfully on win32 client but failed with all web browsers? Have you ever tried Mac and what about Mac?
2. your addin's functionalities are all good on win32 client?
3. does the browser show anything when it failed to install?
4. when publishing the addin, nothing wrong happened?

**Comment by nifty-nev** [comment]
Hi @RuizhiSunMS,
1. Yes, it failed all web browsers including Safari on Mac.
2. Yes, our addin's functionalities are all good on win32 client.
3. The browser provided conole errors as per the content of the attached zip file.  Nothing seemed specific to the addin's failure to preinstall.  The feedback ribbon at the bottom of the word window briefly flashes "Installing add-ins"...
4. Publishing all seemed to go well.

In stepping through js source during an attempted preinstall, I have noted that storeType is consistently set to '10' ('excatalog').  Do I need to do something to set it to OMEX or WOPIcatalog?

Thanks

**Comment by RuizhiSunMS** [comment]
@nifty-nev , thx for what you shared. From now on, I suppose your case to be about installation rather than office api. I would reach out to related colleagues. 
Create an item #9490450 into our backlog to track it. We will reply to you as soon as there is any progress. Thank you for your patience.

**Comment by guoms1** [comment]
Hi, @nifty-nev 

I’m currently trying to reproduce the issue you’re experiencing, but I’ve noticed that I’m missing some key details. Could you please clarify the following?

1. Which add-in are you encountering the issue with?  
2. How was it preinstalled?  
3. Does it require any special configuration, such as logging in with a test account?  

It would be really helpful if you could share detailed repro steps to guide us through reproducing the issue. Even better, if possible, a short repro steps video would greatly assist us in reproducing and ultimately resolving the problem more efficiently.

Thank you so much for your help, and I look forward to your response!

**Comment by nifty-nev** [comment]
Hi @guoms1, thanks for following up.  I recently resolved this issue by stepping through minified office.js code to confirm required file discovery settings by locating pertinent runtime checks.  Doing this, I identified 'AllowExternalMarketPlace' as an essential setting in this case.  Once I set that to be true in our checkFileInfo response, the add-in loaded successfully.  Surprisingly, inclusion of two other settings of potential interest, 'EnableWopiCatalog' and 'AppCatalogUrl', appeared to make no difference to the pre-install.  Thanks again for your time, case closed as far as I am concerned.
